### ISSUE TITLE
Negative sampling / candidate sampling with Keras API

### ISSUE THREAD

**Issue by Santosh-Gupta** [tensorflow/tensorflow#34131]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

It looks like TF Keras has no built in functions to do candidate sampling like in TF 1.0. As far as I can tell, the official keras word2vec example doesn't have negative sampling https://www.tensorflow.org/tutorials/text/word_embeddings even though it's an integral part for producing quality word embeddings. 

**Will this change the current api? How?**

The original tensorflow losses had options for negative sampling, including different distributions to select from. So perhaps some new loss functions and candidate sampling options will need to be added. 

**Who will benefit with this feature?**

Anyone who is training embedding models.

**Comment by tanzhenyu** [comment]
@Santosh-Gupta What do you mean by "The original tensorflow losses had options for negative sampling, including different distributions to select from" whereas TF2 doesn't have it?

**Comment by tanzhenyu** [comment]
Closing this. If the above article doesn't help you, let us know!
